controllers/datafoundationclient.go

```diff
@@ -208,3 +234,28 @@ func (r *dataFoundationClientReconciler) reconcileDefaultFSStorageClassClaim() e
 
 	return nil
 }
+
+func (r *dataFoundationClientReconciler) hasOCSVolumes() (bool, error) {
+	storageClassList := storagev1.StorageClassList{}
+	if err := r.list(&storageClassList); err != nil {
+		return false, fmt.Errorf("failed to list Storage Classes: %v", err)
+	}
+	ocsStorageClass := make(map[string]bool)
+	for i := range storageClassList.Items {
+		storageClassName := &storageClassList.Items[i].Name
+		if *storageClassName == defaultBlockStorageClassClaimName || *storageClassName == defaultFSStorageClassClaimName {
+			ocsStorageClass[*storageClassName] = true
+		}
+	}
```

Comment:
- declaring the var as pointer seems better considering the rest of the operations 🤔 
```suggestion
	storageClassList := &storagev1.StorageClassList{}
	if err := r.list(storageClassList); err != nil {
		return false, fmt.Errorf("failed to list Storage Classes: %v", err)
	}
	ocsStorageClass := make(map[string]bool)
	for i := range storageClassList.Items {
		storageClassName := storageClassList.Items[i].Name
		if storageClassName == defaultBlockStorageClassClaimName || storageClassName == defaultFSStorageClassClaimName {
			ocsStorageClass[storageClassName] = true
		}
	}
```

Reply:
updated